llvm/test/CodeGen/AIE/aie2/schedule/interblock/liveness-region.mir

```diff
@@ -0,0 +1,248 @@
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LIVENESS-LABEL: liveness_regions:BB0 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: (empty)
+
+  bb.0:
+    liveins: $r1, $r2, $r5
+    $r0 = nsw ADD $r1, $r2, implicit-def $srcarry
+    $r3 = nsw ADD $r0, $r1, implicit-def $srcarry
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $r3, implicit-def $r5
+    $r4 = nsw ADC $r3, $r5, implicit-def $srcarry, implicit $srcarry
+    RET implicit $lr
+    DelayedSchedBarrier implicit $r4
+...
+
+
+
+#     bb.0
+#     /  \
+# bb.1    bb.2
+#     \  /
+#     bb.3
+---
+name:            liveness_branch_regions
+alignment:       16
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LIVENESS-LABEL: liveness_branch_regions:BB0 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $r2 $r4 $r3
+  ; CHECK-LIVENESS-NEXT: liveness_branch_regions:BB1 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $lr $r5 $r2
+  ; CHECK-LIVENESS-NEXT: liveness_branch_regions:BB2 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $lr $r5 $r2
+  ; CHECK-LIVENESS-NEXT: liveness_branch_regions:BB3 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: (empty)
+
+  bb.0:
+    liveins: $r1, $r2, $r5
+    successors: %bb.1, %bb.2
+    $r0 = nsw ADD $r1, $r2, implicit-def $srcarry
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $r0, implicit-def $r1
+    $r3 = nsw ADD $r0, $r1, implicit-def $srcarry
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $r3, implicit-def $r5
+    $r4 = nsw ADC $r3, $r5, implicit-def $srcarry, implicit $srcarry
+    $r7 = MOV_mv_scl $srcarry
+    JNZ $r7, %bb.2
+    DelayedSchedBarrier
+  bb.1:
+    successors: %bb.3
+    liveins: $r2, $r4
+    $r5 = nsw ADD $r2, $r4, implicit-def $srcarry
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $r5
+    $r5 = ABS $r5, implicit-def $srcarry
+    J_jump_imm %bb.3
+    DelayedSchedBarrier
+  bb.2:
+    successors: %bb.3
+    liveins: $r3, $r4
+    $r1 = nsw ADD $r3, $r4, implicit-def $srcarry
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $r1
+    $r2 = MUL_mul_r_rr $r1, $r1
+  bb.3:
+    liveins: $r2, $r5, $r6
+    $r1 = nsw ADD $r5, $r2, implicit-def $srcarry
+    RET implicit $lr
+    DelayedSchedBarrier implicit $r1 
+...
+
+#     bb.0
+#       |
+#     bb.1* (self loop)
+#       | 
+#     bb.2
+---
+name:            liveness_loop_regions
+alignment:       16
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LIVENESS-LABEL: liveness_loop_regions:BB0 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $bml0 $amll0 $amlh0 $bml2 $amll2 $amlh2 $r1 $crfpmask
+  ; CHECK-LIVENESS-NEXT: liveness_loop_regions:BB1 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $bml0 $amll0 $amlh0 $bml2 $amll2 $amlh2 $r1 $crfpmask $lr
+  ; CHECK-LIVENESS-NEXT: liveness_loop_regions:BB2 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: (empty)
```

Comment:
The log is a bit confusing, `$bml2` is live in `bb.2`, but I assume what we are dumping are the block liveouts? Maybe drop the `Live Registers:` part on the second line and just list the registers?

Reply:
This cannot be done, as `->dump()` is only way to dump the LiveRegs. `Live Registers:` is logged by inbuilt `dump()`